src/main/java/net/neoforged/neoforge/common/extensions/ILevelReaderExtension.java

```diff
@@ -37,4 +38,6 @@ default <T> Holder<T> holderOrThrow(ResourceKey<T> key) {
     default <T> Optional<Holder.Reference<T>> holder(ResourceKey<T> key) {
         return this.self().registryAccess().holder(key);
     }
+
+    FlagManager getModdedFlagManager();
```

Comment:
I believe this needs to be `default` to avoid breaking `LevelReader` implementations written against vanilla.

Reply:
Hmm, would the following be an acceptable default implementation?
It is a none null method which must lookup a valid `FlagManager`, default implementations grab from either the client or server.

I originally didn't give it a body, so that i knew all the places to implement it in, so should be fine and never throw under normal circumstances.

```suggestion
    default FlagManager getModdedFlagManager() {
        throw new NotImplementedException("ILevelReaderExtension must be implemented to lookup a valid FlagManager");
    }
```

Reply:
Additionally shouldn't I also add the same kind of body to `IMinecraftExtension` and `MinecraftServerExtension`?
So that vanilla has default implementations.

Reply:
I have gone ahead and made the above suggested change, if a empty body is preferred please let me know and i will remove the thrown exception.